clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonClusterProvider.groovy

```diff
@@ -228,32 +229,11 @@ class AmazonClusterProvider implements ClusterProvider<AmazonCluster> {
     Collection<CacheData> images = cacheView.getAll(IMAGES.ns, allImages.keySet())
     images.each { image ->
       def serverGroupIds = allImages[image.id]
-      Map buildInfo = null
-
-      String appVersionTag = image.attributes.tags?.find { it.key == "appversion" }?.value
-      if (appVersionTag) {
-        def appVersion = AppVersion.parseName(appVersionTag)
-        if (appVersion) {
-          buildInfo = [package_name: appVersion.packageName, version: appVersion.version, commit: appVersion.commit] as Map<Object, Object>
-          if (appVersion.buildJobName) {
-            buildInfo.jenkins = [name: appVersion.buildJobName, number: appVersion.buildNumber]
-          }
-          def buildHost = image.attributes.tags.find { it.key == "build_host" }?.value ?: defaultBuildHost
-          if (buildHost && buildInfo.containsKey("jenkins")) {
-            ((Map) buildInfo.jenkins).host = buildHost
-          }
-          def buildInfoUrl = image.attributes.tags?.find { it.key == "build_info_url" }?.value ?: null
-          if (buildInfoUrl) {
-            buildInfo.buildInfoUrl = buildInfoUrl
-          }
-
-        }
-      }
 
       serverGroupIds.each { serverGroupId ->
         def serverGroup = serverGroups[serverGroupId]
         serverGroup.image = image.attributes
-        serverGroup.buildInfo = buildInfo
+        serverGroup.buildInfo = getBuildInfoFromImage(image)
```

Comment:
this retains the original behavior of setting the `buildInfo` to `null` if any of the conditions in the original code are met such that the value to be set is `null`.